### ISSUE TITLE
Leftover AWS resources after stratus warmup error

### ISSUE THREAD

**Issue by rollwagen** [DataDog/stratus-red-team#293]
**What is not working?**

Leftover created AWS resources that are not transparent to the stratus user.

Running `stratus warmup aws.persistence.lambda-overwrite-code` yielded the following error message:
```
Error: creating IAM Role (stratus-red-team-olc-lambda-aqunhnbzot): InvalidClientTokenId: The security token included in the request is invalid
	status code: 403, request id: 9d8a81f5-ab17-4936-a8ca-3cfc12c58425

  with aws_iam_role.lambda-update,
  on main.tf line 31, in resource "aws_iam_role" "lambda-update":  31: resource "aws_iam_role" "lambda-update" {
```

Underlying root case is that stratus was run in a session that uses temporary credential via aws-vault i.e. `aws-vault exec <profile_name>`. The limitations are well-known and documented
[Temporary credentials limitations with STS, IAM](https://github.com/99designs/aws-vault/blob/master/USAGE.md#temporary-credentials-limitations-with-sts-iam)

Stratus warmup, however, only failed after it had already created some resources. In this case e.g. a S3 bucket. These resources are not cleaned-up and also not necessarily transparent for the stratus user with the error message above. Also `stratus status aws.persistence.lambda-overwrite-code` showed 'COLD'.

Suggestion would be to change [runner.go](https://github.com/DataDog/stratus-red-team/blob/main/v2/pkg/stratus/runner/runner.go#L78-L80) so that a cleanup is attempted when a warmup failure happens, so that there are no leftoever 'hidden' resources:

```go
    if err != nil {
        log.Println("Error during warming up, trying to clean up first.")
        _ = m.TerraformManager.TerraformDestroy(m.TerraformDir)
        return nil, errors.New("unable to run terraform apply on prerequisite: " + errorMessageFromTerraformError(err))
    }
```

**What OS are you using?**. Mac OS X

**What is your Stratus Red Team version?**.  Current HEAD i.e. `42c6b41`

**Full output?** _see output above_

**Files in $HOME/.stratus-red-team?**
```
/Users/user/.stratus-red-team/aws.persistence.lambda-overwrite-code:
drwxr-xr-x user staff  96 B  Fri Jan 20 18:32:43 2023  .terraform
.rw-r--r-- user staff   0 B  Fri Jan 20 18:32:45 2023  .terraform-initialized
.rw-r--r-- user staff 2.4 KB Fri Jan 20 18:32:45 2023  .terraform.lock.hcl
.rw-r--r-- user staff 2.2 KB Fri Jan 20 18:32:37 2023  main.tf
.rw-r--r-- user staff 6.7 KB Fri Jan 20 18:32:53 2023  terraform.tfstate
```

Looking forward to feedback. Happy to do the PR for this too.

**Comment by christophetd** [comment]
Thanks for the report! I think the behavior you suggests makes sense, happy to accept a PR!

**Comment by christophetd** [comment]
closing as this was solved a while ago by #315
